frontend/src/components/Toaster/index.tsx

```diff
@@ -0,0 +1,166 @@
+	content?: React.ReactNode
+}
+
+type Options = {
+	id?: string
+	duration?: number
+	content?: React.ReactNode
+}
+
+const DEFAULT_DURATION = 1500
+
+const toastVar = makeVar<Toast[]>([])
+
+const makeToast = (message: string, type: Toast['type'], options?: Options) => {
+	const duration = options?.duration || DEFAULT_DURATION
+	const id = options?.id || Math.random().toString(36).substring(2, 9)
+	return { id, message, type, duration, content: options?.content }
+}
+
+const addToast = (toast: Toast) => {
+	const toasts = toastVar()
+	toastVar([...toasts, toast])
+
+	return new Promise<void>((resolve) => {
+		setTimeout(() => {
+			destroy(toast.id)
+			resolve()
+		}, toast.duration)
+	})
+}
+
+const destroy = (id: string) => {
+	const toasts = toastVar()
+	toastVar(toasts.filter((t) => t.id !== id))
+}
+
+const error = (message: string, options?: Options) => {
+	const toast = makeToast(message, ToastType.error, options)
+	return addToast(toast)
+}
+
+const info = (message: string, options?: Options) => {
+	const toast = makeToast(message, ToastType.info, options)
+	return addToast(toast)
+}
+
+const success = (message: string, options?: Options) => {
+	const toast = makeToast(message, ToastType.success, options)
+	return addToast(toast)
+}
+
+const warning = (message: string, options?: Options) => {
+	const toast = makeToast(message, ToastType.warning, options)
+	return addToast(toast)
+}
+
+export const toast = {
+	destroy,
+	error,
+	info,
+	success,
+	warning,
+}
+
+export const Toaster: React.FC = () => {
+	const toasts = useReactiveVar(toastVar)
+	return (
+		<Stack cssClass={styles.toast} pr="32" pb="32" gap="16">
+			{toasts.map((toast) => (
+				<ToastItem key={toast.id} toast={toast} />
+			))}
+		</Stack>
+	)
+}
+
+const ICON_TYPE_MAPPINGS = {
+	[ToastType.error]: {
+		icon: <IconSolidXCircle size={14} color="#CD2B31" />,
+	},
+	[ToastType.info]: {
+		icon: <IconSolidInformationCircle size={14} color="#6F6E77" />,
+	},
+	[ToastType.success]: {
+		icon: <IconSolidCheckCircle size={14} color="#18794E" />,
+	},
+	[ToastType.warning]: {
+		icon: <IconSolidExclamation size={14} color="#AD5700" />,
+	},
+}
+
+type ToastItemProps = {
+	toast: Toast
+}
+
+const ToastItem: React.FC<ToastItemProps> = ({ toast }) => {
+	const typeInfo = ICON_TYPE_MAPPINGS[toast.type]
+
+	const handleClose = (e: React.MouseEvent) => {
+		e.stopPropagation()
+		destroy(toast.id)
+	}
+
+	return (
+		<Box
+			backgroundColor="white"
+			gap="6"
+			p="8"
+			borderRadius="6"
+			border="divider"
+			display="flex"
+			style={{ width: '280px' }}
+		>
+			<Box display="flex" alignItems="flex-start" pt="2">
+				{typeInfo.icon}
+			</Box>
+			<Stack direction="row" justifyContent="space-between" width="full">
+				<Stack direction="column" gap="8">
+					<Box display="flex" alignItems="flex-start" pt="4">
+						<Text weight="bold">{toast.message}</Text>
+					</Box>
+
+					{!!toast.content && (
+						<Box display="flex">{toast.content}</Box>
+					)}
+				</Stack>
```

Comment:
Quick thought on the spacing in here... Wonder if we want to wrap `toast.content` in `Text` as well so we can control the spacing more precisely. Also wondering if the button will be too close to the text if there is no `content` right now.

Finally, in a text-only toast that is multiline it feels like the spacing on the bottom is off since it's less that the top.

<img width="343" alt="image" src="https://github.com/highlight/highlight/assets/308182/4bfa14f2-4501-4b8d-8d1f-c27fefb46f89">

Looking at the mocks, it appears we have the same padding on the top and the bottom.

<img width="1084" alt="Screenshot 2024-05-22 at 2 55 07 PM" src="https://github.com/highlight/highlight/assets/308182/70a64e83-13be-4589-a713-3e91ec2e1b53">

Should that `pt="4"` actually be `py="4"`?

Reply:
Also, the button in this case is added through the `content`. Not sure if that changes anything

Reply:
![Screenshot 2024-05-22 at 5 06 19 PM](https://github.com/highlight/highlight/assets/17744174/08e31287-c4af-40a7-9a08-3475f73878a6)